src/Symfony/Component/Config/Definition/ArrayNode.php

```diff
@@ -29,7 +29,7 @@ class ArrayNode extends BaseNode implements PrototypeNodeInterface
     protected $addIfNotSet = false;
     protected $performDeepMerging = true;
     protected $ignoreExtraKeys = false;
-    protected $removeExtraKeys = true;
+    protected $removeExtraKeys = false;
```

Comment:
This is not BC: if someone was ignoring values, you would not remove values anymore by default